src/core_plugins/kibana/public/dashboard/reducers/view.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Reducer } from 'redux';
+import { ViewActions, ViewActionTypeKeys } from '../actions';
+
+import { Filters, Query, TimeRange } from 'ui/embeddable';
+import { QueryLanguageType } from 'ui/embeddable/types';
+import { DashboardViewMode } from '../dashboard_view_mode';
+import { PanelId, ViewState } from '../selectors';
+
+const setVisibleContextMenuPanelId = (view: ViewState, panelId: PanelId) => ({
+  ...view,
+  visibleContextMenuPanelId: panelId,
+});
+
+const updateHidePanelTitles = (view: ViewState, hidePanelTitles: boolean) => ({
+  ...view,
+  hidePanelTitles,
+});
+
+const minimizePanel = (view: ViewState) => ({
+  ...view,
+  maximizedPanelId: undefined,
+});
+
+const maximizePanel = (view: ViewState, panelId: PanelId) => ({
+  ...view,
+  maximizedPanelId: panelId,
+});
+
+const updateIsFullScreenMode = (
+  view: ViewState,
+  isFullScreenMode: boolean
+) => ({
+  ...view,
+  isFullScreenMode,
+});
+
+const updateTimeRange = (view: ViewState, timeRange: TimeRange) => ({
+  ...view,
+  timeRange,
+});
+
+const updateFilters = (view: ViewState, filters: Filters) => ({
+  ...view,
+  filters,
+});
+
+const updateQuery = (view: ViewState, query: Query) => ({
+  ...view,
+  query,
+});
+
+const updateUseMargins = (view: ViewState, useMargins: boolean) => ({
+  ...view,
+  useMargins,
+});
+
+const updateViewMode = (view: ViewState, viewMode: string) => ({
+  ...view,
+  viewMode,
+});
+
+export const viewReducer: Reducer<ViewState> = (
+  view: ViewState = {
+    filters: [],
+    hidePanelTitles: false,
+    isFullScreenMode: false,
+    maximizedPanelId: undefined,
```

Comment:
__nit:__ if you apply same thing as I mentioned above (`?` optional properties) you won't need to define `undefine` properties.

Reply:
cooool